exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -142,6 +146,14 @@ type MasterAuthorizedNetworksConfigCidrBlock struct {
 	CidrBlock string `json:"cidr_block,omitempty"`
 }
 
+// NetworkPolicy represents configuration options for NetworkPolicy feature of the GKE cluster.
+type NetworkPolicy struct {
+	// The selected network policy provider.
+	// +kubebuilder:validation:Enum=calico
+	// +optional
+	Provider string `json:"provider,omitempty"`
```

Comment:
If NetworkPolicy is supplied by the user should we require that they supply **Provider**. It feels like they should.